vista_2d/components.py

```diff
@@ -0,0 +1,352 @@
+
+logger = get_logger("VistaCell")
+
+
+class LoadTiffd(MapTransform):
+    def __call__(self, data):
+        d = dict(data)
+        for key in self.key_iterator(d):
+            filename = d[key]
+
+            extension = os.path.splitext(filename)[1][1:]
+            image_size = None
+
+            if extension in ["tif", "tiff"]:
+                img_array = tifffile.imread(filename)  # use tifffile for tif images
+                image_size = img_array.shape
+                if len(img_array.shape) == 3 and img_array.shape[-1] <= 3:
+                    img_array = np.transpose(
+                        img_array, (2, 0, 1)
+                    )  # channels first without transpose
+            else:
+                img_array = np.array(
+                    PIL.Image.open(filename)
+                )  # PIL for all other images (png, jpeg)
+                image_size = img_array.shape
+                if len(img_array.shape) == 3:
+                    img_array = np.transpose(img_array, (2, 0, 1))  # channels first
+
+            if len(img_array.shape) not in [2, 3]:
+                raise ValueError(
+                    "Unsupported image dimensions, filename "
+                    + str(filename)
+                    + " shape "
+                    + str(img_array.shape)
+                )
+
+            if len(img_array.shape) == 2:
+                img_array = img_array[np.newaxis]  # add channels_first if no channel
+
+            if key == "label":
+                if img_array.shape[0] > 1:
+                    print(
+                        f"Strange case, label with several channels {filename} shape {img_array.shape}, keeping only first"
+                    )
+                    img_array = img_array[[0]]
+
+            elif key == "image":
+                if img_array.shape[0] == 1:
+                    img_array = np.repeat(
+                        img_array, repeats=3, axis=0
+                    )  # if grayscale, repeat as 3 channels
+                elif img_array.shape[0] == 2:
+                    print(
+                        f"Strange case, image with 2 channels {filename} shape {img_array.shape}, appending first channel to make 3"
+                    )
+                    img_array = np.stack(
+                        (img_array[0], img_array[1], img_array[0]), axis=0
+                    )  # this should not happen, we got 2 channel input image
+                elif img_array.shape[0] > 3:
+                    print(
+                        f"Strange case, image with >3 channels,  {filename} shape {img_array.shape}, keeping first 3"
+                    )
+                    img_array = img_array[:3]
+
+            meta_data = {
+                ImageMetaKey.FILENAME_OR_OBJ: filename,
+                ImageMetaKey.SPATIAL_SHAPE: image_size,
+            }
+            d[key] = MetaTensor.ensure_torch_and_prune_meta(img_array, meta_data)
+
+        return d
+
+
+class SaveTiffd(MapTransform):
```

Comment:
Didn't see where use this class, can we remove it?

Reply:
Okay removed.